providers/curation/github.js

```diff
@@ -34,6 +34,69 @@ class GitHubCurationService {
     }
   }
 
+  _updateContent(coordinates, currentContent, newContent) {
+    // ensure the coordinates are set
+    currentContent.coordinates = {
+      type: coordinates.type,
+      provider: coordinates.provider,
+      namespace: coordinates.namespace === '-' ? null : coordinates.namespace,
+      name: coordinates.name
+    }
+
+    // remove and re-add the revisions property so it appears after the metadata properties (and default it to empty)
+    // and merge in the new values overwriting current revisions as needed. The patch content coming in should be
+    // relative to the current merged content, not that which may happen to be in this PR
+    const revisions = currentContent.revisions
+    delete currentContent.revisions
+    currentContent.revisions = _.assign(revisions || {}, newContent)
+
+    // return the serialized YAML
+    return yaml.safeDump(currentContent, { sortKeys: true })
+  }
+
+  // merge and write the new content for the given coordinates on the identified branch. This may write multiple
+  // revisions for the same component. Data for existing revisions is completely overwritten if present in the new content.
+  async _writePatch(coordinates, newContent, branch) {
+    const { owner, repo } = this.options
+    const currentContent = await this.getAll(coordinates)
+    const updatedContent = this._updateContent(coordinates, currentContent, newContent)
+    const content = base64.encode(updatedContent)
+    const path = this._getCurationPath(coordinates)
+    const message = `Update ${path} ${coordinates.revision}`
+
+    // if the current content came from Git, get the SHA and use that to update the content. Otherwise, add a new file
+    if (currentContent._origin)
+      return github.repos.updateFile({ owner, repo, path, message, content, branch, sha: currentContent._origin.sha })
+    return github.repos.createFile({ owner, repo, path, message, content, branch })
+  }
+
+  async addOrUpdate(github, coordinates, patch) {
+    if (!patch.patch)
+      throw new Error('Cannot add or update an empty patch. Did you forget to put it in a "patch" property?')
+    const { owner, repo, branch } = this.options
+
+    // Get the sha for the current master and create a branch for the PR
+    const master = await github.repos.getBranch({ owner, repo, branch: `refs/heads/${branch}` })
+    const sha = master.data.commit.sha
+    const prBranch = this._getBranchName(coordinates)
+    await github.gitdata.createReference({ owner, repo, ref: `refs/heads/${prBranch}`, sha })
+
+    // write all the patch files
+    await Promise.all(
```

Comment:
If I'm reading this right, we'll end up with 1 commit per change. Obviously this can be squashed when merging PRs, but it does make for large histories. Is this fine?